### PR TITLE
Upgrade reflectable to use new version of analyzer

### PR DESCRIPTION
**Pull request by eernstg**
This PR changes reflectable such that it uses the analyzer version 6.4.0, and also updates other dependencies.

This includes several changes to the code associated with evaluation of constant expressions (where the new version of the analyzer does not support the approach which was used for this purpose previously).

### CODE REVIEW

**Comment by matt-hall-zory** [comment]
It looks like you have path ^1.9.0 set as the minimum version.
However, this will rule out use of integration_test which is pinned by flutter to ^1.8.3

**Comment by eernstg** [comment]
Thanks for the heads-up, @matt-hall-zory! I'll adjust the version to 1.8.3.

PS: I'm waiting for a feature that would restore the ability to evaluate constants using analyzer 6.3.0, so I can't land this one right now.

**Review by Sten435 — APPROVED** [review]

**Comment by eernstg** [comment]
We still don't have "official" support for constant evaluation in the analyzer, but I've taken a shortcut: Importing some `src` libraries. This is not guaranteed to continue to work (because breaking changes in `src` libraries are not breaking ;-), but it will get us on track again.

@sigurdm, would you be able to take a look at this PR?

**Review by stan-at-work — APPROVED** [review]

**Comment by eernstg** [comment]
@stan-at-work and @Sten435, thanks for the reviewing votes!

For this package (or anything in github.com/google and similar locations), I do need an approving review from a Google employee as well. So that's the reason why this hasn't landed yet.

File: reflectable/lib/src/builder_implementation.dart
Comment (sigurdm) [inline reflectable/lib/src/builder_implementation.dart:5588]: Is this something that needs to be addressed before landing?

Otherwise we should probably file an issue and format this as
```suggestion
  // TODO(https://github.com/google/reflectable/issues/<issue number>) Analyzing without errors until this point.
```

**Review by sigurdm — APPROVED** [review]

File: reflectable/lib/src/builder_implementation.dart
Comment (eernstg) [reply reflectable/lib/src/builder_implementation.dart:5588]: Yes, certainly, `!!!TODO!!!` should never be committed. Thanks for catching that!

Comment (eernstg) [reply reflectable/lib/src/builder_implementation.dart]: Changed the function to `async`, now using `_severe` like everywhere else. No more TODOs. ;-)

**Comment by eernstg** [comment]
:tada: Unstuck!

**Comment by codekeyz** [comment]
Amazing work @eernstg

**Pull request merged**
